### PR TITLE
[CEN-1273] Add endpoint to expose PGP public key for trx encryption

### PR DESCRIPTION
**Pull request by alberto-re-pagopa**
This PR adds a new endpoint under the RTD CSV Transaction API to allow the instances of Batch Acquirer to fetch at runtime the PGP key to use for the encryption of transaction files before the upload.
With this API it will no longer be needed to manually download the key and place it in a path local to the BA, thus simplifying the key distribution process.

### List of changes

- New /rtd/csv-transaction/publickey method (GET)

### Motivation and context

- Allow the clients to programmatically get the correct key to use for encryption

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [x] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/api.tf [hunk 1]
```diff
@@ -440,7 +440,6 @@ module "rtd_csv_transaction" {
   api_management_name = module.apim.name
```
Comment (GiovanniMancini) [inline src/api.tf:440]: Maybe we need a `count` meta-argument to exclude production environment. I think it's preferable to harden the blob storage before moving to production. What's your opinion?

**Review by GiovanniMancini — COMMENTED** [review]
Maybe we are not ready to go to production, since we still have security issues on blob storage

File: src/api.tf (hunk 1 above)
Comment (alberto-re-pagopa) [reply src/api.tf:440]: Agree. I'll add a commit to include this feature.

**Review by pasqualedevita — DISMISSED** [review]
lock removed.

not blocking but using a managed identity to write into blob storage it's a better approach
here the example
https://github.com/pagopa/pagopa-infra/pull/113/files

**Comment by alberto-re-pagopa** [comment]
> lock removed.
> 
> not blocking but using a managed identity to write into blob storage it's a better approach here the example https://github.com/pagopa/pagopa-infra/pull/113/files

Thanks for the suggestion @pasqualedevita. Just to be sure: you're not referring to the content of this PR but to the create-sas-token-policy policy which uses  a Shared Key access to create containers under the blobstorage on-the-fly, right? (https://github.com/pagopa/cstar-infrastructure/blob/main/src/api/rtd_csv_transaction/create-sas-token-policy.xml.tpl from line 69 to 163)

**Pull request merged**